implementations/code_df/issue.py

```diff
@@ -0,0 +1,61 @@
+from metric import Metric
+import utils
+
+
+class Issue(Metric):
+    """
+    Initilizes self.df, the DataFrame, with one issue per row.
+
+    :param items: A list of dictionaries.
+        Each item is a Perceval dictionary, obtained from a JSON
+        file or from Perceval directly.
+
+    :param date_range: A tuple which represents the period of interest
+        It is of the form (since, until), where since and until are
+        strings. Either, or both can be None. If, for example, since
+        is None, that would mean that all issues from the first issue
+        to the issue which last falls inside the until range will be
+        included.
+    """
+
+    def __init__(self, items, date_range=(None, None)):
+
+        self.since, self.until = date_range
+        super().__init__(items)
+
+    def _flatten(self, item):
+        """
+        Flatten a raw issue fetched by Perceval into a flat dictionary.
+
+        A list with a single flat directory will be returned.
+        That dictionary will have the elements we need for computing metrics.
+        The list may be empty, if for some reason the issue should not
+        be considered.
+
+        :param item: raw item fetched by Perceval (dictionary)
+        :returns:   list of a single flat dictionary
+        """
+
+        creation_date = utils.str_to_date(item['data']['created_at'])
+        if self.since and (self.since > creation_date):
+            return []
+
+        if self.until and (self.until < creation_date):
+            return []
+    
+        # all pull requests are considered to be issues by default.
+        # this check removes pull requests from the
+        # analysis.
+        if 'pull_request' in item['data']:
+            return []
```

Comment:
This condition is to avoid analyzing pull requests in the data. All pull requests are treated as issues and hence the issues data also includes pull requests.

Reply:
I think we can assume all items are pull requests. In other words, we can assume that the filtering is done before the data arrives into this class. So instead of including this check, let's include a comment in the description of the class stating that it should be initialized with a list of pull requests.

Reply:
> I think we can assume all items are pull requests.

@jgbarah, this is an issue heirarchy metric and hence, should be used on issue data, not pull requests. I think you meant to say this.

> So instead of including this check, let's include a comment in the description of the class stating that it should be initialized with a list of pull requests.

I think you mean to say, "... that it should be initialized with a list of issues." I think this will be difficult to do since the only differences between regular pull requests and those pull requests in the issue data fetched by Perceval are a few fields. 

Please just confirm what you meant and I'll remove the condition to **remove pull requests and not issues** and add a comment as required. Thanks!

Reply:
Oooops. Yes. Let's assume Perceval items are filtered before they are fed to this class, and all of them are issues. So, we wouldn't need to filter them when initlalizing the object. Sorry for the noise.

The code for filtering in only issues is simple:

```
issues = [item if pull_request not in item['data'] for item in items]
```
(code not tested, but should be like that)